src/legacyApi.ts

```diff
@@ -0,0 +1,315 @@
+/**
+ * This module previously defined an exported API for the extension. While
```

Comment:
Would you be able to help us clean this up?  I fear that if it doesn't happen as the new API comes in, this legacy will stick around forever.

Reply:
I started doing that, but I'm concerned I might accidentally break something in the process. The extension appears to have dependencies on these types even when they aren't used for API access, and there are cases where functions are dynamically accessed by name (like command registration in setup) so language services will have a difficult time determining that there are no regressions. Do you think the tests are in a good enough state that passing tests after a refactor is enough to ensure quality?

Reply:
I think most of the types survive.  They can move to TS modules where it makes sense, or we just have a new `types.ts` file where we can dump a bunch of stuff.  Just the old CMakeToolsAPI interface gets renamed to CMakeProject (or something) and we delete some of the functions/properties on it.

We currently only have `export class CMakeProject implements api.CMakeToolsAPI {` that I would consider to be a concern, but if we rename the interface as suggested above, not much actually changes.  You have CMakeToolsApiImpl which will be the real implementation of the new API and is the entry point to get a CMakeProject that they can work with.

Reply:
Legacy API is removed in the latest update. Most types went to `cache.ts`, `cmakeProject.ts`, and `cmakeDriver.ts`. I only refactored the types to remove `legacyApi.ts`, so any stale logic is still present.